enhancements/network/openstack-dual-stack.md

```diff
@@ -0,0 +1,175 @@
+subnets that were pre-filled for the machine-api-provider-openstack to create one unique Port
+with both addresses configured for the Server.
+6. The CIDRs of the subnets have to added to the `cloud-provider-config.yaml`
+in the `address-sort-order` field to make sure IPv4 address will have higher priority in the
+Node addresses.
+7. The installer will validate the subnets and VIPs, add the security group to the api and ingress
+ports and link the floating IPs to those Ports. Also, create the additional IPv6 security group rules
+and configured the servers with dual-stack addresses and add the dual-stack allowed address pairs.
+8. In machine-config-operator, kubelet will be update to not be configured with node-ips.
+9. cloud-provider-openStack will filter the node addresses that are meant to be specified in the Status of the Node
+by looking for a tag that was applied in the main Ports of the Node.
+
+### API Extensions
+
+The `machinesSubnet` field, which is available in the `install-config.yaml` under
+`platform` and `openstack` will be deprecated in favor of a new field named `machinesSubnets`
+to persist both IPv4 and IPv6 provided Subnets.
+
+For installation on OpenStack with dual-stack kubelet will no longer rely on the node-ip to get the node
+addresses, given that when using external cloud provider, the node-ip currently has no support for dual-stack, this [enhancement](https://github.com/kubernetes/enhancements/pull/3706) describes it. Instead,
+the node addresses will be provided directly by the cloud provider openstack, which can rely on a tag to filter them. This [issue](https://github.com/kubernetes/cloud-provider-openstack/issues/2096) details the idea behind it.
+
+### Implementation Details/Notes/Constraints
+
+The idea behind using bring your own network instead of basic IPI installation is to allow support
+for a wider variety of ways on how to configure the Network and Subnet. As an example, the Ports have
+to be pre-created by the user because when using SLAAC or Stateless the specification of fixed-ips during
+Port creation is not allowed. UPI can also be an option in future releases.
+
+### Risks and Mitigations
+
+We need to validate if the proposal for not setting the node-ip will be a robust enough
+solution that we want to maintain, specially around upgrades. If not robust, we can use the dual-stack support for
+node-ip that is being [developed upstream](https://github.com/kubernetes/kubernetes/pull/116305).
+
+### Drawbacks
+
+NA
+
+## Design Details
+
+### Open Questions
+
+1. Do we want to let the user configure the `MachineConfig` prior installation according
+ to the type of IPv6 mode used or we want to teach the machine-config-operator how to
+ configure those?
+2. How does the routing work when using tenant network and attempt to connect to the API
+ using IPv6 address in the GUA range? until now we tested with Provider network.
```

Comment:
IMO this should be up to the user to figure that out, since we do BYON here, we expect the routing to be configured before the deployment of OCP.

Reply:
good point. I believe we should give some sort of guidelines for routing configuration in the documentation, and also test this scenario right?